### PR TITLE
[TE] add enpoints to save dimension filter for RCA v2 POC

### PR DESCRIPTION
**Pull request by vincentchenjl**
This PR is add endpoints that support storing dimensional filters per metric level for RCA v2 POC.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/v2/rootcause/AbstractRCAModuleConfig.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * under the License.
+ */
+
+package org.apache.pinot.thirdeye.dashboard.resources.v2.rootcause;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.io.Serializable;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public abstract class AbstractRCAModuleConfig implements Serializable {
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/v2/rootcause/AbstractRCAModuleConfig.java:26]: add some java doc about this class?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datalayer/pojo/RootcauseTemplateBean.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.List;
+import java.util.Map;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RootcauseTemplateBean extends AbstractBean {
+  String name;
+  String owner;
+  String application;
+  long metricId;
+  List<Map<String, Object>> modules;
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datalayer/pojo/RootcauseTemplateBean.java:32]: Will it work if the type is defined as `List<AbstractRCAModuleConfig>`? it should automatically convert it into Json when saving?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/v2/rootcause/AbstractRCAModuleConfig.java (hunk 1 above)
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/v2/rootcause/AbstractRCAModuleConfig.java:26]: Ad some java doc to this class

Comment (vincentchenjl) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/v2/rootcause/AbstractRCAModuleConfig.java:30]: Added comment.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datalayer/pojo/RootcauseTemplateBean.java (hunk 2 above)
Comment (vincentchenjl) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datalayer/pojo/RootcauseTemplateBean.java:32]: The reason I use `Map<String, Object>` is that this map includes more than just`AbstractRCAModuleConfig`. Once we start working on modularizing RCA components, this map should include class names for modules and corrsponding configurations.

**Pull request merged**